src/rules.jl

```diff
@@ -217,6 +217,38 @@ function apply!(o::Adam, state, x, dx)
   return (mt, vt, βt .* β), dx′
 end
 
+"""
+    Lion(η = 0.001, β::Tuple = (0.9, 0.999))
+
+[Lion](https://arxiv.org/abs/2302.06675) optimiser.
+
+# Parameters
+- Learning rate (`η`): Amount by which gradients are discounted before updating
+                       the weights.
+- Decay of momentums (`β::Tuple`): Exponential decay for the first (β1) and the
+                                   second (β2) momentum estimate.
+"""
+struct Lion{T} <: AbstractRule
+  eta::T
+  beta::Tuple{T,T}
+end
+Lion(η = 1f-3, β = (9f-1, 9.99f-1)) = Lion{typeof(η)}(η, β)
+
+init(o::Lion, x::AbstractArray) = zero(x)
+
+function apply!(o::Lion, state, x, dx)
+  η, (β1, β2) = o.eta, o.beta
+  β2 > 1 && error("")
```

Comment:
This was a placeholder just because I dropped a factor `(1-β2)` inside the `sign`. Maybe it should say something more friendly:
```suggestion
  (β1 < 1 && β2 < 1) || throw(DomainError("Lion optimiser requires that both decay rates β be less than 1, but got β1 = $β1, β2 = $β2"))
```